From a196f16cfa891096554701d8a0e9c02a2448c5f3 Mon Sep 17 00:00:00 2001
From: Filippo Luca Ferretti <filippoluca.ferretti@outlook.com>
Date: Wed, 1 Apr 2026 17:55:14 +0200
Subject: [PATCH] Add missing string include in nvtx_utils.h

In abseil-cpp 20260107, absl/strings/string_view.h no longer
transitively includes <string>. nvtx_utils.h uses std::string
(std::stack<std::string>) but only includes <stack> and
absl/strings/string_view.h, so std::string is incomplete.

diff --git a/third_party/xla/xla/backends/profiler/gpu/nvtx_utils.h b/third_party/xla/xla/backends/profiler/gpu/nvtx_utils.h
--- a/third_party/xla/xla/backends/profiler/gpu/nvtx_utils.h
+++ b/third_party/xla/xla/backends/profiler/gpu/nvtx_utils.h
@@ -17,6 +17,7 @@
 #define XLA_BACKENDS_PROFILER_GPU_NVTX_UTILS_H_
 
 #include <stack>
+#include <string>
 
 #include "absl/strings/string_view.h"
 #include "tsl/platform/macros.h"
